EventSubscriber/RequestSubscriber.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Ise\WebSecurityBundle\EventSubscriber;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class RequestSubscriber implements EventSubscriberInterface
+{
+    private $active;
+    private $corsEndpoints;
+    public function __construct(ContainerInterface $container)
+    {
+        $this->active = $container->getParameter('ise_security.fetch_metadata.active');
+        $this->active = []; //Stub until Config is constructed
+    }
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['requestEvent', 0],
+            ]
+        ];
+    }
+
+    public function requestEvent(RequestEvent $event): void
+    {
+            $request = $event->getRequest();
+            if($this->active) {
+                if(!$this->allowRequest($request)){
+                    $response = new Response('', Response::HTTP_UNAUTHORIZED);
+                    $response->headers->set('Vary', 'sec-fetch-site');
+                    $event->setResponse($response);
+                }
+            }
+    }
+    /**
+     * Method to check the current request against base Fetch Metadata security guidelines.
+     * Guide based on https://web.dev/fetch-metadata/ article
+     *
+     * @param Request $req 
+     * @return void
+     */
+    private function allowRequest(Request $req) 
```

Comment:
`allowRequest` seems to generic here. Isn't this just *an* example of a potential policy you would want to use for allowing/denying a request based on the `sec-fetch`*` headers?

Reply:
Absolutely too generic name wise, will be broken into a separate class possibly too. But from what I've been reading of the implementation guides that policy is the minimum of what should be included